File: py/features.py
# FederalRegister SDK feature factory

from feature.base_feature import FederalRegisterBaseFeature
from feature.test_feature import FederalRegisterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FederalRegisterBaseFeature(),
        "test": lambda: FederalRegisterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()

File: rb/features.rb
# FederalRegister SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FederalRegisterFeatures
  def self.make_feature(name)
    case name
    when "base"
      FederalRegisterBaseFeature.new
    when "ratelimit"
      FederalRegisterRatelimitFeature.new
    when "retry"
      FederalRegisterRetryFeature.new
    when "test"
      FederalRegisterTestFeature.new
    when "timeout"
      FederalRegisterTimeoutFeature.new
    else
      FederalRegisterBaseFeature.new
    end
  end
end
